### ISSUE TITLE
请求提供一份简单样例

### ISSUE THREAD

**Issue by MotorImagery** [Paranioar/UniPT#3]
能否提供一个关于CNN和transformer的简单样例代码，并在其上使用UniPT，以便于可以通过学习这个样例来将该方法使用在其他领域上。十分感谢！

**Comment by Paranioar** [comment]
The definitions of UniPT on CNN and Transformer are in  [adapter_for_cnn.py](https://github.com/Paranioar/UniPT/blob/main/GPO/lib/adapter_for_cnn.py) and [adapter_for_transformer.py](https://github.com/Paranioar/UniPT/blob/main/GPO/lib/adapter_for_transformer.py), respectively.

Please refer to two types of **self.side_tuning_module** in [GPO/lib/encoders.py](https://github.com/Paranioar/UniPT/blob/main/GPO/lib/encoders.py) to activate them. 
Don't forget to adjust related configurations including hidden_size, num_layer, and etc for your domains.
Besides, please follow the forward function in [EncoderImageFull and EncoderText](https://github.com/Paranioar/UniPT/blob/main/GPO/lib/encoders.py) to implement the UniPT.

Some tips might be useful to you.
1) Set the learning rate of UniPT module as 10x of pretrained weights. (or 5x, or 20x)
2) Set the weight decay from 1e-4 to 1e-2. Smaller sizes of pretrain model, larger values of WD.
3) For CNN, the outputs of first one or two shallow layers can be abandoned. Generally, the memory usage is largely reduced, and performance is not sacrificed.
4) The structure of FFN layer in the confidence module can be adjusted (deeper or shallower) to get better results.
5) The orders of ReLU and Normlayer (LayerNorm or BatchNorm) are sometimes different according to some references.
6) Please set the minimum value of **eps** as 1e-6 in case of FP16 or Mixed FP16+32. It bothers me before！
7) Other issues ......

*I'm really sorry that I've been so busy lately. Hence, I'll be out of contact for a while until the end of September.*
